lib/asm-parser.js

```diff
@@ -591,6 +591,7 @@ export class AsmParser extends AsmRegex {
             if (match) {
                 maybeRemovePreviousLabel();
                 mayRemovePreviousLabel = filters.libraryCode;
+                source = null;
```

Comment:
This is not correct for most compilers and assembly. Perhaps overriding this class and function would be better than to cause bugs with this.